tests/integration_tests/run_no_reprocessing/test.py

```diff
@@ -1,47 +1,59 @@
-from saltproc import DepcodeSerpent
-from saltproc import Simulation
-from saltproc import Reactor
-from pyne import serpent
+"""Run SaltProc without reprocessing"""
 import os
 import glob
-import sys
+from pathlib import Path
+
 import numpy as np
 import pytest
 
-path = os.path.realpath(__file__)
-sys.path.append(os.path.dirname(os.path.dirname(path)))
-
-# global class object
-directory = os.path.dirname(path)
-
-sss_file = directory + '/int_test'
-iter_matfile = directory + '/int_test_mat'
-db_file = directory + '/db_test.h5'
-
-depcode = DepcodeSerpent(
-    exec_path='sss2',
-    template_input_file_path=directory +
-    '/saltproc_9d.inp',
-    geo_files=[
-        os.path.join(
-            directory,
-            '../../test_geo.inp')],
-    npop=100,
-    active_cycles=20,
-    inactive_cycles=5)
-depcode.iter_inputfile = sss_file
-depcode.iter_matfile = iter_matfile
-
-simulation = Simulation(sim_name='Integration test',
-                        sim_depcode=depcode,
-                        core_number=1,
-                        node_number=1,
-                        db_path=db_file)
-
-tap = Reactor(volume=1.0,
-              power_levels=[1.250E+09],
-              dep_step_length_cumulative=[3])
+from pyne import serpent
+from saltproc import app
+from saltproc import DepcodeSerpent, Simulation, Reactor
+
+
+@pytest.fixture
+def setup():
+    cwd = Path(__file__).parents[0].resolve().as_posix()
+    main_input = cwd + '/test_input.json'
+
+    input_path, process_input_file, path_input_file, object_input = \
+        app.read_main_input(main_input)
+
+    depcode = app._create_depcode_object(object_input[0])
+    sss_file = cwd + '/_test'
+    depcode.iter_inputfile = sss_file
+    depcode.iter_matfile = cwd + '/_test_mat'
+
+    simulation = app._create_simulation_object(object_input[1], depcode, 1, 1)
+
+    reactor = app._create_reactor_object(object_input[2])
+
+    return cwd, simulation, reactor, sss_file
+
+
+@pytest.mark.slow
+def test_integration_2step_saltproc_no_reproc_heavy(setup):
+    cwd, simulation, reactor, sss_file = setup
+    runsim_no_reproc(simulation, reactor, 2)
+    saltproc_out = sss_file + '_dep.m'
+
+    ref_result = serpent.parse_dep(cwd + '/reference_dep.m', make_mats=False)
+    test_result = serpent.parse_dep(saltproc_out, make_mats=False)
+
+    ref_mdens_error = np.loadtxt(cwd + '/reference_error')
+
+    ref_fuel_mdens = ref_result['MAT_fuel_MDENS'][:, -2]
+    test_fuel_mdens = test_result['MAT_fuel_MDENS'][:, -1]
+
+    test_mdens_error = np.array(ref_fuel_mdens - test_fuel_mdens)
+    np.testing.assert_array_equal(test_mdens_error, ref_mdens_error)
+    # Cleaning after testing
+    out_file_list = glob.glob(cwd + '/_test*')
+    for file in out_file_list:
+        try:
+            os.remove(file)
+        except OSError:
+            print("Error while deleting file : ", file)
 
 
 def runsim_no_reproc(simulation, reactor, nsteps):
```

Comment:
should this function be defined before `test_integration_2step_saltproc_no_reproc_heavy` since it is called from there?

Reply:
Hmm, I don't think it matters that much. This style of function ordering is what OpenMC uses.

Reply:
What is the style?

Reply:
Define top-level functions first, then define any other functions in the order they are called.